os/kernel/sched/sched_unlock.c

```diff
@@ -125,9 +125,14 @@ int sched_unlock(void)
 		int cpu = this_cpu();
 
 		/* Decrement the preemption lock counter */
-
 		if (rtcb->lockcount > 0) {
+			DEBUGASSERT(g_cpu_schedlock == SP_LOCKED && \
+					(g_cpu_lockset & (1 << cpu)) != 0);
+
 			rtcb->lockcount--;
+		} else {
+			DEBUGASSERT(g_cpu_schedlock == SP_UNLOCKED && \
+					(g_cpu_lockset & (1 << cpu)) == 0);
```

Comment:
Hi @kishore-sn , how about moving this to line 146~148?

Reply:
Additionally, the spin_clrbit in line 150~152 is redundant, since we will already face assert crash if the condition is violated.

Reply:
After thinking again, I think there should be no cases to enter sched_unlock with lockcount = 0, so it should be fine with current implementation~
Sorry for mistake, please ignore the comments above~